core/src/main/java/org/infinispan/configuration/cache/QueryConfiguration.java

```diff
@@ -12,9 +12,10 @@
 public class QueryConfiguration extends ConfigurationElement<QueryConfiguration> {
 
    public static final AttributeDefinition<Integer> DEFAULT_MAX_RESULTS = AttributeDefinition.builder(org.infinispan.configuration.parsing.Attribute.DEFAULT_MAX_RESULTS, 100).immutable().build();
+   public static final AttributeDefinition<Integer> HIT_COUNT_ACCURACY = AttributeDefinition.builder(org.infinispan.configuration.parsing.Attribute.HIT_COUNT_ACCURACY, 10_000).immutable().build();
```

Comment:
I am personally leaning towards Integer.MAX_VALUE as a default value since it requires no explaining as the hit count will always be correct. Having a lower value means the user has to know that the hit may be correct if it is less than the hit count. Any other value returned really means there may be greater than hit count accuracy elements, which requires additional logic processing for their code by default.

Reply:
The idea is with Infinispan 15 to return a flag to say if the hit count is exact or not => https://github.com/infinispan/infinispan/pull/11014.
With Infinispan 14 since we cannot change the API we return the hit count as null in case the count is not exact.
In any case I think that we make the user aware.

I'm in favor of limiting the total hit count by default (to a conveniently large value) in order to avoid in case of large indexes by default to traverse all the index in order to collect the query result.

Reply:
I think that this is the same idea of having [accurate-size](https://github.com/fax4ever/infinispan/blob/2709ee7decc2523631c1dcb796fcbfccd77b9441/core/src/main/resources/schema/infinispan-config-15.0.xsd#L811) defaulted to false.
It is not perfect I know, but maybe a little more user friendly for non advanced users.